backend/dataall/core/environment/services/environment_service.py

```diff
@@ -447,8 +465,62 @@ def add_consumption_role(session, uri, data=None) -> (Environment, EnvironmentGr
             permissions=permissions.CONSUMPTION_ROLE_ALL,
             resource_type=ConsumptionRole.__name__,
         )
+
+        EnvironmentService._generate_managed_policies_for_consumption_role(environment, consumption_role)
+
         return consumption_role
 
+    @staticmethod
+    def _generate_managed_policies_for_consumption_role(environment, consumption_role):
```

Comment:
+1 to the above - I think it is important we make this feature more generic rather than requiring ConsumptionRole() to be created for each EnvGroup Role now. Also in the future if we extend this feature of `dataallManaged` to other IAM entities I think it would get very confusing

Reply:
I put it all in separate environment-related service